component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPKeyGenerator.java

```diff
@@ -55,54 +55,87 @@
-                    generatedSecretKey = key.getKey();
-                    claims.put(TOTPAuthenticatorConstants.SECRET_KEY_CLAIM_URL,
-                            TOTPUtil.encrypt(generatedSecretKey));
-                } else {
-                    decryptedSecretKey = TOTPUtil.decrypt(storedSecretKey);
-                }
-                if (StringUtils.isNotEmpty(generatedSecretKey)) {
-                    secretKey = generatedSecretKey;
-                } else {
-                    secretKey = decryptedSecretKey;
-                }
-
-                String issuer = TOTPUtil.getTOTPIssuerDisplayName(tenantDomain, context);
-                String displayUsername = TOTPUtil.getTOTPDisplayUsername(tenantAwareUsername);
-                String qrCodeURL =
-                        "otpauth://totp/" + issuer + ":" + displayUsername + "?secret=" + secretKey + "&issuer=" +
-                                issuer + "&period=" + timeStep;
-                encodedQRCodeURL = Base64.encodeBase64String(qrCodeURL.getBytes());
-                claims.put(TOTPAuthenticatorConstants.QR_CODE_CLAIM_URL, encodedQRCodeURL);
+                claims = getGeneratedClaims(username, tenantDomain, storedSecretKey, refresh, timeStep, context);
             }
         } catch (UserStoreException e) {
             throw new TOTPException(
                     "TOTPKeyGenerator failed while trying to get the user store manager from user realm of the user : " +
                             tenantAwareUsername, e);
+        } catch (AuthenticationFailedException e) {
+            throw new TOTPException(
+                    "TOTPKeyGenerator cannot find the property value for encoding method", e);
+        }
+        return claims;
+    }
+
+    /**
+     * Generate TOTP secret key for federated user, encoding method and QR Code url for user.
+     *
+     * @param username        Username of the user.
+     * @param storedSecretKey Stored secret key.
+     * @param refresh         Boolean type of refreshing the secret token.
+     * @param context         Authentication context.
+     * @return Generated TOTP related claims of federated user.
+     * @throws TOTPException when user realm is null or while decrypting the key
+     */
+    public static Map<String, String> generateClaimsForFedUser(String username, String tenantDomain,
+                                                               String storedSecretKey, boolean refresh,
+                                                               AuthenticationContext context)
+            throws TOTPException {
+
+        long timeStep = getTimeStamp(context, username);
+        Map<String, String> claims =
+                getGeneratedClaims(username, tenantDomain, storedSecretKey, refresh, timeStep, context);
+        return claims;
+    }
+
+    public static Map<String, String> getGeneratedClaims(String username, String tenantDomain, String storedSecretKey,
```

Comment:
Let's add method level comments to the public methods.

Reply:
Fixed https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/94/commits/01e2f562f9e60129c3810e75f2cb8f16020632bc